core/src/main/java/org/apache/accumulo/core/conf/Property.java

```diff
@@ -1531,21 +1547,19 @@ public static boolean isClassProperty(String key) {
     propertiesByKey = new HashMap<>();
     validPrefixes = new HashSet<>();
     validProperties = new HashSet<>();
+    validTableProperties = new HashSet<>();
 
     for (Property p : Property.values()) {
+      propertiesByKey.put(p.getKey(), p);
       if (p.getType().equals(PropertyType.PREFIX)) {
         validPrefixes.add(p.getKey());
       } else {
+
         validProperties.add(p.getKey());
-      }
-      propertiesByKey.put(p.getKey(), p);
-    }
 
-    validTableProperties = new HashSet<>();
-    for (Property p : Property.values()) {
-      if (!p.getType().equals(PropertyType.PREFIX)
-          && p.getKey().startsWith(Property.TABLE_PREFIX.getKey())) {
-        validTableProperties.add(p.getKey());
+        if (p.getKey().startsWith(Property.TABLE_PREFIX.getKey())) {
+          validTableProperties.add(p.getKey());
+        }
```

Comment:
I like that this logic was moved into the previous loop to avoid looping twice... although I don't think it will make much of a difference since it's a static initializer that will only run once and there aren't that many to loop through.

However, it's a problem that the logic has changed to no longer exclude PREFIX types that start with "table." This structure should not contain prefix types, but only actual keys. We don't want users to be able to set `table.custom.` or `table.constraint.` by themselves because these are incorrectly deemed to be valid properties when they aren't.

Reply:
Should be fixed in update